### PR TITLE
Add git-lfs package.

### LINKED ISSUE #1636: New Package: git-lfs
Has anyone ever looked at adding a [git-lfs](https://git-lfs.github.com) package?

Specifically, for the client side support. This is a missing piece for LLNL LC users that might be thinking about using BitBucket's Git LFS support.

### PR DESCRIPTION
**Pull request by tgamblin**
Resolve #1636.

@IanLee1521 @xjrc @davydden

### CODE REVIEW

**Comment by davydden** [comment]
LGTM

File: var/spack/repos/builtin/packages/git-lfs/package.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+class GitLfs(Package):
+    """Tool for managing large files with Git."""
+
+    homepage = "https://git-lfs.github.com"
+    url      = "https://github.com/github/git-lfs/archive/v1.4.1.tar.gz"
+
+    version('1.4.1', 'c62a314d96d3a30af4d98fa3305ad317')
+
+    depends_on('go@1.5:', type='build')
+    depends_on('git@1.8.2:', type='run')
```
Comment (IanLee1521) [inline var/spack/repos/builtin/packages/git-lfs/package.py]: Does this mean a minimum or required version `1.8.2` ?

Comment (davydden) [inline var/spack/repos/builtin/packages/git-lfs/package.py]: correct.

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/git-lfs/package.py]: Minimum. `go@1.5` means it requires 1.5. `go@1.5:` means it requires at least version 1.5. Think of it like Python list indexing.

**Comment by tgamblin** [comment]
@xjrc @IanLee1521: are you working on this one?  I can just merge it and let you submit a PR later...

**Comment by xjrc** [comment]
@tgamblin: I'd say just merge it and I'll get around to trying it out and updating it sometime soon.  Thanks again for your help with this!

**Pull request merged**